mctreesearch4j/src/main/kotlin/mctreesearch4j/AdvancedStatelessSolver.kt

```diff
@@ -0,0 +1,43 @@
+package mctreesearch4j
+
+import kotlin.math.ln
+import kotlin.math.sqrt
+
+open class AdvancedStatelessSolver<TState, TAction>(
```

Comment:
I don't think we should ship this in the library. I don't think capturing the exploration term is a common enough scenario to be included by default. This is something that should be implemented by the user.

Reply:
I was actually trying to access the internals of the Kotlin implementation from Scala as originally intended. In general its many times more difficult difficult to alter the solver from outside Kotlin, given all the type conversions and syntax issues.

I think I can take out the exploration term extraction, but keep this `AdvancedStatelessSolver`'s `getOptimalHorizon` method or move it into the original solver?